Writerside/topics/raptor.md

```diff
@@ -203,4 +166,64 @@ In the context of the RAPTOR algorithm, a **Route** refers to:
 - **Optimization**: Routes are scanned efficiently, with the algorithm looking at each route at most once per round,
   contributing to RAPTOR's speed and dynamic capabilities.
 
-![raptor-route.png](raptor-route.png)
+![raptor-route.png](raptor-route.png){width="650"}
+
+## The RAPTOR Algorithm Explained in Detail
+
+### Key Features of RAPTOR
+
+**Pareto-Optimality:** RAPTOR computes Pareto-optimal journeys, meaning it considers multiple criteria, such as earliest
+arrival time and the fewest transfers. A journey is Pareto-optimal if no other journey is better in both criteria.
+
+**Round-based Processing:** The algorithm processes journeys in rounds, where each round computes the arrival times for
+journeys that use a specific number of transfers.
+
+**Efficient Route Scanning:** RAPTOR explicitly uses the structure of public transit timetables, scanning each transit
+route
+at most once per round. This ensures that the algorithm is fast and scalable, even for large transit networks.
+
+### How RAPTOR Works
+
+**Initialization:**
+Given a source stop s, a target stop t, and a departure time τ_dep, the algorithm initializes the journey from s.
+
+**Rounds:**
+Each round k computes all journeys with exactly k trips. In the first round, RAPTOR computes journeys with no transfers,
+i.e., direct trips. In subsequent rounds, it computes journeys with one transfer, two transfers, and so on.
+
+![raptor-rounds.png](raptor-rounds.png){width="650"}
+
+**Route Scanning:**
+For each round, the algorithm iterates over every stop on a route and checks whether the current trip improves the
+arrival time at the next stop. This ensures that RAPTOR efficiently computes the earliest arrival time for each stop in
+a transit network.
+Each stop maintains a label indicating its earliest arrival time for the current round. RAPTOR compares these labels and
+updates them if a faster journey is found.
+
+![raptor-scan.png](raptor-scan.png){width="650"}
+
+![route-scan2.png](route-scan2.png){width="650"}
+
+**Stopping Condition:**
+The algorithm terminates when no further improvements in arrival times are possible, ensuring it only performs the
+necessary number of rounds.
+
+### Key Observations
```

Comment:
I think all the key observations are redundant and already explained in the sections above.